invisible_cities/io/mcinfo_io.py

```diff
@@ -325,19 +325,19 @@ def get_event_numbers_in_file(file_name: str) -> np.ndarray:
 
 
 def _get_list_of_events_new(h5in : tb.file.File) -> np.ndarray:
-    mc_tbls  = ['hits', 'particles', 'sns_response']
-    def try_unique_evt_itr(group, itr):
-        for elem in itr:
-            try:
-                yield np.unique(getattr(group, elem).cols.event_id)
-            except tb.exceptions.NoSuchNodeError:
-                pass
-
-    evt_list = list(try_unique_evt_itr(h5in.root.MC, mc_tbls))
-    if len(evt_list) == 0:
-        raise AttributeError("At least one of MC/hits, MC/particles, \
-        MC/sns_response must be present to use get_list_of_events.")
-    return np.unique(np.concatenate(evt_list)).astype(int)
+
+    def get_event_ids_table(tablename):
+        try:
+            evt_list = getattr(h5in.root.MC, tablename).cols.event_id
+        except tb.exceptions.NoSuchNodeError:
+            raise AttributeError('Trying to get event number from MC corrupted file.')
+        return np.unique(evt_list).astype(int)
+   
```

Comment:
It would be good if we could *all* please instruct our editors to remove trailing whitespace whenever we save source files. It's not a big deal, but it helps us keep irrelevant noise out of diffs. (And the most annoying noise usually appears many commits later, and is thrown in the face of someone other than the party that was guilty of introducing the spurious whitespace in the first place ... so it's a matter of courtesy to our colleagues!)

Reply:
Changed